torchbiggraph/converters/importers.py

```diff
@@ -219,14 +236,15 @@ def generate_edge_path_files(
         entities_by_type[rconfig.rhs].num_parts for rconfig in relation_configs
     )
 
-    print(f"- Edges will be partitioned in {num_lhs_parts} x {num_rhs_parts} buckets.")
+    log(f"- Edges will be partitioned in {num_lhs_parts} x {num_rhs_parts} buckets.")
 
     processed = 0
     skipped = 0
-
     # We use an ExitStack in order to close the dynamically-created edge appenders.
     with ExitStack() as appender_stack:
         appenders: Dict[Tuple[int, int], AbstractEdgeAppender] = {}
+        data: Dict[Tuple[int, int], List[Tuple[int, int, int]]] = {}
```

Comment:
Related to the comment above: what if this was already a `Dict[Tuple[int, int], torch.LongTensor]`? So that we didn't have to further convert from `List[Tuple[...]]` to tensor?

Reply:
Yeah, this would be slightly better except that the way we wrote the code we don't know in advance how big to make this Tensor.

Reply:
Can't it be `n_flush_edges` big?

Reply:
sure, if you then want to separately keep track of how many rows are filled. I think the point is, this is all still grossly inefficient; once you're iterating over rows in a file one by one in Python (multiple times), this is not going to be the bottleneck.